From 58872d162bd610578e048bf2832defc93a284a75 Mon Sep 17 00:00:00 2001
From: Giovanni Campagna <gcampagna@src.gnome.org>
Date: Sun, 31 Mar 2013 18:05:49 +0000
Subject: AppDisplay/FrequentView: filter out hidden applications

Filter out all applications which have the NoDisplay, Hidden or
Not/OnlyShowIn bits, as those are not meant to be launched directly.
This also allows the user to filter apps from the frequent view
using alacarte.

https://bugzilla.gnome.org/show_bug.cgi?id=696949
---
diff --git a/js/ui/appDisplay.js b/js/ui/appDisplay.js
index 22ae481..bcfb57a 100644
--- a/js/ui/appDisplay.js
+++ b/js/ui/appDisplay.js
@@ -316,6 +316,8 @@ const FrequentView = new Lang.Class({
     loadApps: function() {
         let mostUsed = this._usage.get_most_used ("");
         for (let i = 0; i < mostUsed.length; i++) {
+            if (!mostUsed[i].get_app_info().should_show())
+                continue;
             let appIcon = new AppIcon(mostUsed[i]);
             this._grid.addItem(appIcon.actor, -1);
         }
--
cgit v0.9.2
